asynciterator.ts

```diff
@@ -652,21 +654,24 @@ export class ArrayIterator<T> extends AsyncIterator<T> {
       this._sourceStarted = true;
 
     let item = null;
-    const buffer = this._buffer;
-    if (buffer) {
-      if (buffer.length !== 0)
-        item = buffer.shift() as T;
-      if (buffer.length === 0) {
+    let { _currentIndex, _buffer } = this;
```

Comment:
Underscores are reserved for private fields, we don't use them for local variables. (I think it's fine to just use `this.` below.)

Reply:
done!